stats.go

```diff
@@ -0,0 +1,423 @@
+/*
+   Copyright (C) nerdctl authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"runtime"
+	"sync"
+	"time"
+
+	formatter_stats "github.com/AkihiroSuda/nerdctl/pkg/stats"
+	v1 "github.com/containerd/containerd/metrics/types/v1"
+	v2 "github.com/containerd/containerd/metrics/types/v2"
+	"github.com/containerd/typeurl"
+	"github.com/docker/cli/cli/command/formatter"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+type stats struct {
+	mu sync.Mutex
+	cs []*formatter_stats.Stats
+}
+
+type statsOptions struct {
+	all        bool
+	noStream   bool
+	noTrunc    bool
+	format     string
+	containers []string
+}
+
+var optionsStats = new(statsOptions)
+
+var statsCommand = &cli.Command{
+	Name:      "stats",
+	Usage:     "Display a live stream of container(s) resource usage statistics",
+	ArgsUsage: "[OPTIONS] [CONTAINER...]",
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name:        "all",
+			Aliases:     []string{"a"},
+			Usage:       "Show all containers (default shows just running)",
+			Destination: &optionsStats.all,
+		},
+		&cli.StringFlag{
+			Name:        "format",
+			Usage:       "Pretty-print images using a Go template",
+			Destination: &optionsStats.format,
+		},
+		&cli.BoolFlag{
+			Name:        "no-stream",
+			Usage:       "Disable streaming stats and only pull the first result",
+			Destination: &optionsStats.noStream,
+		},
+		&cli.BoolFlag{
+			Name:        "no-trunc",
+			Usage:       "Do not truncate output",
+			Destination: &optionsStats.noTrunc,
+		},
+	},
+	Action: func(clicontext *cli.Context) error {
+		optionsStats.containers = clicontext.Args().Slice()
+		showAll := len(optionsStats.containers) == 0
+		closeChan := make(chan error)
+		var err error
+
+		if runtime.GOOS == "windows" {
+			fmt.Println("nerdctl do not yet support windows")
```

Comment:
This is not specific to `stats`, and I don't even think it compiles on Windows, so this Println is not needed